roles/servicetelemetry/templates/manifest_prometheus.j2

```diff
@@ -9,6 +9,7 @@ metadata:
 spec:
 {% if observability_strategy != "use_community" %}
   version: null
```

Comment:
"Specifying `spec.version` is still necessary to ensure the Prometheus Operator knows which version of Prometheus is being configured."[1]

I assume this will eventually cause problems, but since it's existing code I'm willing to let it go for the purposes of this PR

[1] https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#commonprometheusfields

Reply:
I think the version field here is to help the Operator know what fields / configuration are valid, and the default is to use the configuration for the latest upstream version (which would have been in sync without the image being specified). We don't really use anything super special, so I think this is relatively low risk, but is still a bit of a risk in case the Prometheus configuration changes in a way that isn't backwards compatible at some point.

I guess we could hope that whatever version COO is using would be compatible with the Prometheus controller being provided by COO?

Reply:
FWIW everything passed testing as it stands now, so hopefully that'll continue to work going forward.